driver/stub/stub_usbd.c

```diff
@@ -244,41 +249,48 @@ select_usb_conf(usbip_stub_dev_t *devstub, USHORT bVal)
 	PURB		purb;
 	PUSBD_INTERFACE_LIST_ENTRY	pintf_list;
 	NTSTATUS	status;
-
+	struct _URB_SELECT_CONFIGURATION	*purb_selc;
+	
 	dsc_conf = get_usb_dsc_conf(devstub, (UCHAR)bVal);
 	if (dsc_conf == NULL) {
 		DBGE(DBG_GENERAL, "select_usb_conf: non-existent configuration descriptor: index: %hu\n", bVal);
 		return FALSE;
 	}
 
 	pintf_list = build_default_intf_list(dsc_conf);
+	if (pintf_list == NULL) {
+		DBGE(DBG_GENERAL, "select_usb_conf: out of memory: pintf_list\n");
+		ExFreePoolWithTag(dsc_conf, USBIP_STUB_POOL_TAG);
+		return FALSE;
+	}
 
 	status = USBD_SelectConfigUrbAllocateAndBuild(devstub->hUSBD, dsc_conf, pintf_list, &purb);
 	if (NT_ERROR(status)) {
 		DBGE(DBG_GENERAL, "select_usb_conf: failed to selectConfigUrb: %s\n", dbg_ntstatus(status));
 		ExFreePoolWithTag(pintf_list, USBIP_STUB_POOL_TAG);
+		ExFreePoolWithTag(dsc_conf, USBIP_STUB_POOL_TAG);
 		return FALSE;
 	}
 
 	status = call_usbd(devstub, purb);
-	if (NT_SUCCESS(status)) {
```

Comment:
Is there any reason for following code changes? They look similar to old codes.

Reply:
Mostly for legibility but also to catch an unhandled out of memory condition with pintf_list and an a failure case that forgot to free dsc_conf on the second failure case.

Old form was:
1. catch fail -> free and exit
2. catch fail -> free and exit
3. catch fail -> free and exit
4. all good -> free and exit
5. catch fail -> free and exit

I felt the function made more sense as:
1. catch fail -> free and exit
2. catch fail -> free and exit
3. catch fail -> free and exit
4. catch fail -> free and exit
5. all good -> free and exit

Reply:
Ok.. your work looks better! 😄